pkg/combustion/templates/13-embedded-registry.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -euo pipefail
+
+mkdir -p /etc/hauler
+
+mv haul.tar.zst /etc/hauler/haul.tar.zst
```

Comment:
I don't think that `/etc/hauler/` is the right place to store the Hauler archive. Can we consider moving this to `/var/lib/` please? That's where the majority of the other mutable data lives, and is not subject to btrfs snapshot rollbacks on SLE Micro.

Reply:
Will update 👍

Reply:
After testing, I think this won't work as `/var/lib/` doesn't seem to be available during combustion

Reply:
What about `/opt`? @rdoxenham

Reply:
Ignore this, you've already done it.

Reply:
Yep @dbw7 and I spoke about this offline. `/opt/hauler/` works well.